src/ADAL.PCL/Internal/Telemetry/ApiEvent.cs

```diff
@@ -0,0 +1,144 @@
+        {
+            Tenant = PlatformPlugin.CryptographyHelper.CreateSha256Hash(tenantId);
+            SetEvent(EventConstants.Tenant, Tenant);
+
+            if (userinfo != null)
+            {
+                Idp = userinfo.IdentityProvider;
+                SetEvent(EventConstants.Idp, Idp);
+
+                DisplayableId = PlatformPlugin.CryptographyHelper.CreateSha256Hash(userinfo.DisplayableId);
+                SetEvent(EventConstants.DisplayableId, DisplayableId);
+
+                UserId = PlatformPlugin.CryptographyHelper.CreateSha256Hash(userinfo.UniqueId);
+                SetEvent(EventConstants.UserId, UserId);
+            }
+
+            Authority = HashTenantIdFromAuthority(authenticator.Authority);
+
+            SetEvent(EventConstants.Authority, Authority);
+
+            AuthorityType = authenticator.AuthorityType.ToString();
+            SetEvent(EventConstants.AuthorityType, AuthorityType);
+
+            IsDeprecated = false;
+            SetEvent(EventConstants.IsDeprecated, IsDeprecated.ToString());
+
+            SetEvent(EventConstants.ApiId, apiId);
+        }
+
+        internal string Tenant { get; set; }
+
+        internal string Idp { get; set; }
+
+        internal string UserId { get; set; }
+
+        internal string Authority { get; set; }
+
+        internal string AuthorityType { get; set; }
+
+        internal string ValidateAuthority { get; set; }
+
+        internal string DisplayableId { get; set; }
+
+        internal string IdentityProvider { get; set; }
+
+        internal bool IsDeprecated { get; set; }
+
+        internal bool ExtendedExpires { get; set; }
+
+        internal string HashTenantIdFromAuthority(string authority)
+        {
+            Uri uri = new Uri(authority);
+            Regex regex = new Regex(uri.AbsolutePath);
+            string result = regex.Replace(authority, PlatformPlugin.CryptographyHelper.CreateSha256Hash(uri.AbsolutePath), 1);
+            return result;
+        }
+
+        internal override void SetEvent(string eventName, string eventParameter)
+        {
+            if (!string.IsNullOrEmpty(eventParameter))
+            {
+                EventDictitionary[eventName] = eventParameter;
+            }
+        }
+
+        internal void SetExtraQueryParameters(string extraQueryParameter)
+        {
+            string[] result = extraQueryParameter.Split('&');
+            StringBuilder stringbuilder = new StringBuilder();
+            foreach (string s in result)
+            {
+                if (s.Contains("="))
+                {
+                    stringbuilder.Append(s.Split('=')[0]).Append("&");
+                }
+            }
+
+            SetEvent(EventConstants.ExtraQueryParameters,
+                stringbuilder.ToString().Substring(0, stringbuilder.Length - 1));
+        }
+
+        internal override void ProcessEvent(Dictionary<string, string> dispatchMap)
```

Comment:
you are missing
is_successful
api_error_code

not in the doc yet, but was conveyed on slack